src/e2e/puppeteer/__tests__/caret-multiline.ts

```diff
@@ -0,0 +1,254 @@
+    const importText = `
+  - a
+    - b
+      - c
+      - Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable('c')
+    await click(editableNodeHandle, { edge: 'left' })
+
+    await press('ArrowDown')
+    await press('ArrowDown')
+
+    // the focus must be in the middle of the multi-line thought after cursor down
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe(
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.",
+    )
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBeGreaterThan(70)
+  })
+
+  it('on cursorDown, the caret should move from the beginning of a multi-line cursor into the 2nd line of the same cursor.', async () => {
+    const importText = `
+  - a
+    - b
+      - c
+      - Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable(
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.",
+    )
+    await click(editableNodeHandle, { edge: 'left' })
+
+    await press('ArrowDown')
+
+    // the focus must be in the middle of the multi-line cursor after cursor down
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe(
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.",
+    )
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBeGreaterThan(70)
+  })
+
+  it('on cursorDown, the caret should move from near the end of a multi-line cursor to the beginning of a new cursor.', async () => {
+    const multiLineCursor =
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended."
+    const importText = `
+  - a
+    - b
+      - c
+      - ${multiLineCursor}
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable(multiLineCursor)
+    await click(editableNodeHandle, { offset: multiLineCursor.length - 2 })
```

Comment:
Nice. I see how since the browser won't line wrap in the middle of the word "recommended", offset `multilineCursor.length - 2` is guaranteed to be on the last line.